catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -50,10 +62,27 @@
 
     private static final long serialVersionUID = 1L;
 
+    private static final Set<AttributeDescriptor> DEFAULT_ATTRIBUTES = Stream.of(new CoreAttributes(),
+            new SecurityAttributes())
+            .map(MetacardType::getAttributeDescriptors)
+            .flatMap(Set::stream)
+            .collect(Collectors.toSet());
+
+    private static final Collector<AttributeDescriptor, ?, Map<String, AttributeDescriptor>>
+            ATTRIBUTE_COLLECTOR =
+            Collectors.collectingAndThen(Collectors.toMap(AttributeDescriptor::getName,
+                    Function.identity(),
+                    (oldVal, newVal) -> oldVal), ImmutableMap::copyOf);
+
+    /*
+     * Only to be used by the getAttributeDescriptors method.
+     */
+    private Set<AttributeDescriptor> cachedDescriptors;
```

Comment:
actually there was some reason i ended up not using the memoized supplier. so switched the getDescriptors method to return the cached set.